src/app/core/setup/context-aware-dialog/context-aware-dialog.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component } from "@angular/core";
+import { MatButtonModule } from "@angular/material/button";
+import { MatDialogActions, MatDialogContent } from "@angular/material/dialog";
+import { DemoAssistanceDialogComponent } from "../demo-assistance-dialog/demo-assistance-dialog.component";
+
+// Define a shared interface for dialog components
+export interface DemoDialogComponentBase {
+  // Add any shared methods/properties here
+}
```

Comment:
good idea. can we already outline some basic properties?

Reply:
if it's not used anywhere yet, I would tend to remove it for now until we need it. (We sometimes ended up with forgotten duplicate structures :sweat_smile: )